corim/signedcorim.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package corim
+
+import (
+	"crypto"
+	"crypto/rand"
+	"errors"
+	"fmt"
+
+	cose "github.com/veraison/go-cose"
+)
+
+var (
+	ContentType = "application/rim+cbor"
+)
+
+// SignedCorim encodes a signed-corim type (i.e., a COSE Sign1 wrapped CoRIM)
+// with signature and verification methods
+type SignedCorim struct {
```

Comment:
Technically speaking the name of the container on line 21 as SignedCorim
is confusing and incorrect in my view.

It should be named as CorimContainer

As it holds both UnsignedCorim as well as a Signed Corim Message.

WDYT! ?

Reply:
I think it is correct as is: the SignedCorim contains an UnsignedCorim which mimics how a signed-corim wraps an unsigned-corim.